src/main/java/org/littleshoot/proxy/impl/TraceableProxyHttpRequestEncoder.java

```diff
@@ -0,0 +1,29 @@
+package org.littleshoot.proxy.impl;
+
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpRequestEncoder;
+
+public class TraceableProxyHttpRequestEncoder extends HttpRequestEncoder {
+
+  @Override
+  protected void sanitizeHeadersBeforeEncode(HttpRequest msg, boolean isAlwaysEmpty) {
+    HttpHeaders headers = msg.headers();
+
+    String headerVgsCustomerB3TraceId = headers.get(TraceUtils.HEADER_VGS_CUSTOMER_B3_TRACEID);
+
+    // need to remove this header no matter what
+    headers.remove(TraceUtils.HEADER_VGS_CUSTOMER_B3_TRACEID);
+
+    if (headerVgsCustomerB3TraceId == null) {
+      // looks like all tracing header were injected by VGS so removing them
+      headers.remove(TraceUtils.HEADER_X_B3_TRACEID);
+      headers.remove(TraceUtils.HEADER_X_B3_SPANID);
+      headers.remove(TraceUtils.HEADER_X_B3_SAMPLED);
+      headers.remove(TraceUtils.HEADER_X_B3_PARENTSPANID);
+    } else {
+      // preserving "X-B3-TraceId" header
+      headers.set(TraceUtils.HEADER_X_B3_TRACEID, headerVgsCustomerB3TraceId);
```

Comment:
if i'm reading this correctly it looks like we're saying that we do _not_ remove `HEADER_VGS_CUSTOMER_B3_*` headers. is this the desired behavior?

Reply:
we do I was going back and forth on how we will preserve customer headers. We do remove these in https://github.com/verygoodsecurity/LittleProxy/pull/86/files#diff-f079e3781194aaacdf2967b5b9b08ab8bb9ea201e877b1af55be58a040414616R17-R21